hypothesis-python/tests/cover/test_conjecture_engine.py

```diff
@@ -1901,3 +1855,23 @@ def test_function(data):
             prefix_data = runner.cached_test_function(hbytes(n))
             assert prefix_data is Overrun
         assert call_count[0] == 1
+
+
+def test_will_evict_entries_from_the_cache(monkeypatch):
+    monkeypatch.setattr(engine_module, "CACHE_SIZE", 5)
+    count = [0]
+
+    def tf(data):
+        data.draw_bytes(1)
+        count[0] += 1
+
+    runner = ConjectureRunner(tf)
```

Comment:
Whoops, so it does, thanks.

I'm thinking it might be worth modifying that behaviour of the runner to be disabled outside the `run` method. I find I end up bumping into it a lot when doing interesting things with the internals, including this sort of testing.